src/WebJobs.Script.WebHost/App_Start/WebApiConfig.cs

```diff
@@ -93,6 +93,7 @@ private static WebHostSettings GetDefaultSettings(ScriptSettingsManager settings
                 settings.ScriptPath = settingsManager.GetSetting(EnvironmentSettingNames.AzureWebJobsScriptRoot);
                 settings.LogPath = Path.Combine(Path.GetTempPath(), @"Functions");
                 settings.SecretsPath = HttpContext.Current.Server.MapPath("~/App_Data/Secrets");
+                settings.IsSelfHost = true;
```

Comment:
Do all the e2e tests pass with this set? Other of your changes below force an auth level of Admin when self hosted. I'd expect that to cause our e2e auth tests to fail.